### ISSUE TITLE
Would this cover sensor driven hand meshes (e.g. HoloLens)?

### ISSUE THREAD

**Issue by fordacious** [immersive-web/webxr-hand-input#5]
In OpenXR we are discussing hand mesh APIs alongside hand joint APIs.
Devices like the HoloLens generate internal mesh representations of hands based on sensor data.
We may choose to keep this in or out of scope of this module.

**Comment by Manishearth** [comment]
This is interesting. My gut feeling is that this should be a second incubation, but perhaps developed in parallel as much as possible. The OpenXR APIs seem to be pretty distinct as well.

**Comment by Manishearth** [comment]
We could, for now, use this repo as a place to discuss any mesh issues and split it out later. I personally have not thought as much about hand meshes.

**Comment by fordacious** [comment]
> The OpenXR APIs seem to be pretty distinct as well.
Yup, its two different (but parallel and closely related) extensions in OpenXR.

**Comment by AdaRoseCannon** [comment]
This could be very useful for hand occlusion for close up manipulation.

**Comment by cabanier** [comment]
> In OpenXR we are discussing hand mesh APIs alongside hand joint APIs.
> Devices like the HoloLens generate internal mesh representations of hands based on sensor data.

That was proposed in webxr meshing in https://github.com/immersive-web/real-world-geometry

**Comment by Codesmith512** [comment]
Just a stranger's 2 cents, but the Hololens 2 does also give hand joints in addition to a mesh representation, so it'd be great to capture both. For some applications joints are easier to use, but for others a full mesh works better.

Plus, it shouldn't be too difficult to create a library that creates default meshes to feed into the hand-mesh API for when only hand joints are available (though going the other way would be considerably harder).

**Comment by Maksims** [comment]
These are two separate pieces of information. For application logic, joints - are the way to go, to do any gesture detection, AI related stuff, physics, etc.
While mesh - is mostly visual-purpose information. Considering variety of applications and their needs, most will implement bespoke hand meshes, while some might rely on provided, where goal is to have XR-friendly content, without investing too much into XR it self. This definitely will make hand tracking easier to start with.

Also major complexity to consider, is that different engines have their own way to create, store and render meshes. This also applies to their skinning systems. So trying to come up with generic solution that can be utilised by all engines and webgl developers - would be a challenge.

**Comment by cabanier** [comment]
I think we can close this issue since we all see to be in agreement that a different API is needed for meshes.
If anyone disagrees, please reopen the issue.
